usr/lib/common/pkey_utils.c

```diff
@@ -750,6 +754,170 @@ CK_RV pkey_aes_cmac(OBJECT *key_obj, CK_BYTE *message,
     return ret;
 }
 
+/**
+ * Return the KM/KMC/KMAC function code for the given key length and
+ * encrypt/decrypt op.
+ */
+unsigned long get_xts_function_code(CK_ULONG keylen, CK_BYTE encrypt)
+{
+    unsigned long fc;
+
+    switch (keylen) {
+    case 32:
+        fc = ENCRYPTED_AES_XTS_128;
+        break;
+    case 64:
+        fc = ENCRYPTED_AES_XTS_256;
+        break;
+    default:
+        return 0;
```

Comment:
Just make sure that the 0 is handled as an error in calling routines. 0 is a valid function code (all the query functions, e.g. KM_QUERY).